src/math/casadi.hpp

```diff
@@ -68,6 +112,246 @@ namespace Eigen
       return std::numeric_limits<double>::digits10;
     }
   };
+
+
+  namespace internal
+  {
+    /// @brief LLT_Traits for casadi::SX-valued matrices
+    ///
+    /// LLT specialization for matrices of casadi::SX containts in m_matrix the L such as LL'=A,
+    /// regardless of the value of UpLo.
+    /// Therefore we define two identical specializations of LLT_Traits for the two values of UpLo.
+    template<int M, int N, int Options>
+    struct LLT_Traits<Matrix<casadi::SX, M, N, Options>, Lower>
+    {
+      using MatrixType = Matrix<casadi::SX, M, N, Options>;
+      typedef const TriangularView<const MatrixType, Lower> MatrixL;
+      typedef const TriangularView<const typename MatrixType::AdjointReturnType, Upper> MatrixU;
+      static inline MatrixL getL(const MatrixType& m) { return MatrixL(m); }
+      static inline MatrixU getU(const MatrixType& m) { return MatrixU(m.adjoint()); }
+      static bool inplace_decomposition(MatrixType& m)
+      { return llt_inplace<typename MatrixType::Scalar, Lower>::blocked(m)==-1; }
+    };
+
+
+    /// @brief LLT_Traits for casadi::SX-valued matrices
+    ///
+    /// LLT specialization for matrices of casadi::SX containts in m_matrix the L such as LL'=A,
+    /// regardless of the value of UpLo.
+    /// Therefore we define two identical specializations of LLT_Traits for the two values of UpLo.
+    template<int M, int N, int Options>
+    struct LLT_Traits<Matrix<casadi::SX, M, N, Options>, Upper>
+    {
+      using MatrixType = Matrix<casadi::SX, M, N, Options>;
+      typedef const TriangularView<const typename MatrixType::AdjointReturnType, Lower> MatrixL;
+      typedef const TriangularView<const MatrixType, Upper> MatrixU;
+      static inline MatrixL getL(const MatrixType& m) { return MatrixL(m); }
+      static inline MatrixU getU(const MatrixType& m) { return MatrixU(m.adjoint()); }
+      static bool inplace_decomposition(MatrixType& m)
+      { return llt_inplace<typename MatrixType::Scalar, Upper>::blocked(m)==-1; }
+    };
+  }
+
+
+  /// @brief LLT template specialization supporting casadi::SX
+  ///
+  template<int M, int N, int Options, int _UpLo>
+  class LLT<Matrix<casadi::SX, M, N, Options>, _UpLo>
```

Comment:
I do not think that you should specialize all the Eigen::LLT for your type.
You should use partial template specialization of methods instead (see https://en.cppreference.com/w/cpp/language/partial_specialization)

Reply:
What needs to be specialized is
```c++
template<typename InputType>
      LLT& compute(const EigenBase<InputType>& matrix)
```

I would need to do a partial specialization of a template member function of a template class. I am not sure if such thing is possible in C++. At least I tried and my C++ skills were not sufficient to do that.